common/src/main/java/com/cssbham/cssminecraft/common/discord/client/JDADiscordClient.java

```diff
@@ -0,0 +1,100 @@
+
+    private final EventBus eventBus;
+    private final Logger logger;
+    private final String botToken;
+    private final long discordServerId;
+    private final long memberRoleId;
+    private final long bridgeChannelId;
+
+    private JDA jda;
+
+    public JDADiscordClient(EventBus eventBus, Logger logger, String botToken, long discordServerId, long memberRoleId, long bridgeChannelId) {
+        this.eventBus = eventBus;
+        this.logger = logger;
+        this.botToken = botToken;
+        this.discordServerId = discordServerId;
+        this.memberRoleId = memberRoleId;
+        this.bridgeChannelId = bridgeChannelId;
+    }
+
+    @Override
+    public void initialise() {
+        if (null != this.jda) {
+            logger.debug("JDA already initialised, skipping initialisation request");
+            return;
+        }
+
+        logger.debug("Initialising JDA");
+
+        this.jda = JDABuilder.createDefault(
+                        botToken
+                ).setMemberCachePolicy(MemberCachePolicy.ALL)
+                .setChunkingFilter(ChunkingFilter.ALL)
+                .enableIntents(GatewayIntent.GUILD_MEMBERS, GatewayIntent.MESSAGE_CONTENT)
+                .addEventListeners(this)
+                .build();
+    }
+
+
+    @Override
+    public void shutdown() {
+        if (null == this.jda) {
+            logger.debug("JDA already stopped, skipping stop request");
+            return;
+        }
+
+        logger.debug("Shutting down JDA");
+
+        jda.shutdownNow();
+        this.jda = null;
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        logger.debug(String.format("Message received from %s: %s", event.getAuthor().getName(), event.getMessage().getContentRaw()));
+        if (!event.isFromGuild() || event.getChannel().getIdLong() != bridgeChannelId ||
+                event.isWebhookMessage() ||
+                event.getMember() == null ||
+                event.getAuthor().isBot() ||
+                event.getMessage().isEdited()) {
+            return;
+        }
+
+        eventBus.dispatch(new DiscordMessageEvent(
+                event.getMember().getEffectiveName(),
+                event.getMessage().getContentRaw(),
+                event.getMember().getColorRaw()
+        ));
+    }
+
+    @Override
+    public boolean isMember(String identifier) {
+        Guild g = jda.getGuildById(discordServerId);
+        if (g == null) return false;
+        Member m = g.getMembers().stream()
+                .filter(mm ->
+                        (mm.getUser().getName() + "#" + mm.getUser().getDiscriminator()).equalsIgnoreCase(identifier) ||
+                                mm.getUser().getName().equalsIgnoreCase(identifier)
+                ).findFirst().orElse(null);
+        if (m == null) return false;
+        return m.getRoles().stream().anyMatch(r -> r.getIdLong() == memberRoleId);
```

Comment:
I presume this is prehistoric code carried over, but this is Matty style variable naming

Reply:
truly great variable names to be sure

Reply:
actually checking this, if ur gonna re-write this - `.getDiscriminator` is depreciated as well

Reply:
> `.getDiscriminator` is depreciated as well

is it deprecated because there's a new method to get the discriminator, or because Discord are getting rid of them?

Reply:
The docs say:
```
Deprecated
This will become obsolete in the future. Discriminators are being phased out and replaced by globally unique usernames. For more information, see New Usernames & Display Names  .```

Reply:
Hmm... do you know if people still have the numeric discriminators? I don't want to remove support for them if people still have them

Reply:
The api docs make no mention of them being completely removed yet, nor can i find a date by which they were to be completely removed

Reply:
> Starting March 4, 2024, Discord will begin assigning new usernames to users who have not chosen one themselves.

It's been 5 months since this, I reckon it's safe to assume they're gone by now